rpc/src/v1/impls/eth.rs

```diff
@@ -545,6 +546,55 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Box::new(future::done(res))
 	}
 
+	fn proof(&self, address: RpcH160, values: Vec<RpcH256>, num: Trailing<BlockNumber>) -> BoxFuture<EthAccount> {
+		let a: H160 = address.clone().into();
+		let key1 = keccak(a);
+
+		let num = num.unwrap_or_default();
+		let id = match num {
+			BlockNumber::Num(n) => BlockId::Number(n),
+			BlockNumber::Earliest => BlockId::Earliest,
+			BlockNumber::Latest => BlockId::Latest,
+			BlockNumber::Pending => {
+				warn!("`Pending` is deprecated and may be removed in future versions. Falling back to `Latest`");
+				BlockId::Latest
+			}
+		};
+
+		try_bf!(check_known(&*self.client, num.clone()));
+		let res = match self.client.prove_account(key1, id) {
+			Some(p) => Ok(EthAccount {
+				 address : address.into(),
+				 balance : p.1.balance.into(),
+				 nonce : p.1.nonce.into(),
+				 code_hash : p.1.code_hash.into(),
+				 storage_hash : p.1.storage_root.into(),
+				 account_proof: Some( p.0.iter().map(|b| Bytes::new(b.clone())).collect::<Vec<Bytes>>()),
+				 storage_proof: Some( values.iter().map(|storage_index| { 
```

Comment:
Would be better to use `filter_map` here and only append valid proofs IMHO

Reply:
changed it, because even for a not existing `0x0` - storage value  there must a proof, so if we don't get a storage proof we skip these keys. (which actually should never happen, since this is only possible if the account does not exist in which case I currently get at least an empty proof)